sb3_contrib/ars/ars.py

```diff
@@ -0,0 +1,289 @@
+
+from sb3_contrib.ars.policies import ARSPolicy
+from sb3_contrib.common.vec_env.async_eval import AsyncEval
+
+
+class ARS(BaseAlgorithm):
+    """
+    Augmented Random Search: https://arxiv.org/abs/1803.07055
+
+    Original implementation: https://github.com/modestyachts/ARS
+    C++/Cuda Implementation: https://github.com/google-research/tiny-differentiable-simulator/
+    Numpy Implementation: https://github.com/alexis-jacq/numpy_ARS/blob/master/asr.py
+
+    :param policy: The policy to train, can be an instance of ARSPolicy, or a string
+    :param env: The environment to train on, may be a string if registred with gym
+    :param n_delta: How many random pertubations of the policy to try at each update step.
+    :param n_top: How many of the top delta to use in each update step. Default is n_delta
+    :param learning_rate: Float or schedule for the step size
+    :param delta_std: Float or schedule for the exploration noise
+    :param zero_policy: Boolean determining if the passed policy should have it's weights zeroed before training, default True.
+    :param alive_bonus_offset: Constant added to the reward at each step, a value of -1 is used in the original paper
+    :param n_eval_episodes: Number of episodes to evaluate each candidate.
+    :param policy_kwargs: Keyword arguments to pass to the policy on creation
+    :param policy_base: Base class to use for the policy
+    :param tensorboard_log: String with the directory to put tensorboard logs:
+    :param seed: Random seed for the training
+    :param verbose: Verbosity level: 0 no output, 1 info, 2 debug
+    :param device: Torch device to use for training, defaults to "auto"
+    :param _init_setup_model: Whether or not to build the network at the creation of the instance
+    """
+
+    def __init__(
+        self,
+        policy: Union[str, Type[ARSPolicy]],
+        env: Union[GymEnv, str],
+        n_delta: int = 64,
+        n_top: Optional[int] = None,
+        learning_rate: Union[float, Schedule] = 0.05,
+        delta_std: Union[float, Schedule] = 0.05,
+        zero_policy: bool = True,
+        alive_bonus_offset: float = 0,
+        n_eval_episodes: int = 1,
+        policy_kwargs: Optional[Dict[str, Any]] = None,
+        policy_base: Type[BasePolicy] = ARSPolicy,
+        tensorboard_log: Optional[str] = None,
+        seed: Optional[int] = None,
+        verbose: int = 0,
+        device: Union[th.device, str] = "cpu",
+        _init_setup_model: bool = True,
+    ):
+
+        super().__init__(
+            policy,
+            env,
+            learning_rate=learning_rate,
+            tensorboard_log=tensorboard_log,
+            policy_base=policy_base,
+            policy_kwargs=policy_kwargs,
+            verbose=verbose,
+            device=device,
+            supported_action_spaces=(gym.spaces.Box, gym.spaces.Discrete),
+            support_multi_env=True,
+            seed=seed,
+        )
+
+        if self.device != th.device("cpu"):
```

Comment:
If we are forcing the device to CPU should we just remove the device parameter entirely?